components/http-accept.js

```diff
@@ -1,55 +1,49 @@
 // http-accept.js
 
 var _ = require('underscore');
-var body = require('body');
-var noflo = require('noflo');
+var getRawBody = require('raw-body');
+var typer = require('media-typer');
 
 var basenode = require('./base-node');
+var promiseComponent = require('./promise-component');
 
-exports.getComponent = function() {
-    return _.extend(new noflo.Component({
-        outPorts: {
-            out: {
-                description: "Request body as a String",
-                datatype: 'string'
-            },
-            rejected: {
-                description: "HTTP 415 Unsupported Media Type request/response pair {req, res}",
-                datatype: 'object'
-            },
-            accepted: {
-                description: "HTTP 202 Accepted request/response pair {req, res}",
-                datatype: 'object'
-            }
+exports.getComponent = promiseComponent({
+    description: "Extracts the request body as a String and produces a 202 Accepted response. Optionally, also filters on request content type.",
+    icon: 'sign-in',
+    outPorts: {
+        rejected: {
+            description: "HTTP 415 Unsupported Media Type request/response pair {req, res}",
+            datatype: 'object'
         },
-        inPorts: {
-            limit: {
-                description: "Request Content-Length limit",
-                datatype: 'int',
-                process: basenode.on({data: basenode.assign('limit')})
-            },
-            encoding: {
-                description: "Request body character encoding",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('encoding')})
-            },
-            type: {
-                description: "Request Content-Type",
-                datatype: 'string',
-                process: basenode.on({data: basenode.assign('types', basenode.push)})
-            },
-            'in': {
-                description: "HTTP request/response pair {req, res}",
-                datatype: 'object',
-                required: true,
-                process: basenode.on({data: handle})
-            }
+        accepted: {
+            description: "HTTP 202 Accepted request/response pair {req, res}",
+            datatype: 'object'
         }
-    }), {
-        description: "Extracts the request body as a String and produces a 202 Accepted response. Optionally, also filters on request content type.",
-        icon: 'sign-in'
-    });
-};
+    },
+    inPorts: {
+        limit: {
+            description: "Request Content-Length limit",
+            datatype: 'int',
+            ondata: basenode.assign('limit')
+        },
+        encoding: {
+            description: "Request body character encoding",
+            datatype: 'string',
+            ondata: basenode.assign('encoding')
+        },
+        type: {
```

Comment:
I was thinking Content-Length (-> content_length_limit), Accept-Encoding, Content-Type.  But maybe we should try it out more first, both because that renaming would make the port names longer and because you have the port descriptions readily available.   We can leave it as is for the moment anyway.